File: DataStructure/Queue/Queue.py
#This class implements a simple Queue data structure using a list in Python.
class Queue:

    # The constructor initializes the queue data structure.
    def __init__(self) -> None:
        self.items = []
    
    # Returns true if the queue is empty, false it is not.
    def is_empty(self) -> bool:
        return len(self.items) == 0 # type: ignore
    
    # Input item to the queue.
    def push(self, item) -> None:
        self.items.append(item)
    
    # Removes the first item from the queue and returns it.
    def pop(self): # type: ignore
        if self.is_empty():
            raise IndexError("pop from empty queue")
        else:
            return self.items.pop(0)  # type: ignore
    
    # Return the number of items in the queue.
    def size(self) -> int: # type: ignore
        return len(self.items) # type: ignore
    
    # Returns the first item from the queue without removing it.
    def peek(self): # type: ignore
        if self.is_empty():
            raise IndexError("peek from empty queue")
        else:
            return self.items[0] # type: ignore

queue = Queue()
queue.push(1)
queue.push(2)
queue.push(3)
print(queue.pop())  # Output: 1
print(queue.peek())  # Output: 2

File: DataStructure/Queue/Queue.js
//This class implements a simple Queue data structure using an array of JavaScript.
class Queue {
    
    //The constructor initializes an empty array to queue data structure.
    constructor() {
        this.items = [];
    }

    //Returns true if the queue is empty, false if it is not.
    isEmpty() { return this.items.length == 0;}

    //Input an element to the end of the queue.
    push(item) { this.items.push(item); }

    //Romove the front element from the queue and return it.
    pop() {
        if(this.isEmpty()) {
            return null;
        }
        else { return this.items.shift(); }
    }

    //Return size of the queue.
    size() { return this.items.length; }

    //Return the front element of the queue without removing it.
    peek() {
        if(this.isEmpty()) {
            return null;
        }
        else { return this.items[0]; }        
    }
}

queue = new Queue();
queue.push(1);
queue.push(2);
queue.push(3);
console.log(queue.peek()); // Output: 1
console.log(queue.pop());  // Output: 1
console.log(queue.peek()); // Output: 2
